packages/lib/src/components/Card/Card.tsx

```diff
@@ -193,31 +208,9 @@ export class CardElement extends UIElement<CardElementProps> {
                 commonProps={{ isCollatingErrors: this.props.SRConfig.collateErrors }}
             >
                 <ClickToPayProvider clickToPayService={this.clickToPayService}>
```

Comment:
It's not possible to also put the `ClickToPayProvider` inside the `ClickToPayWrapper`?

Reply:
The reason I didn't put it there is because the `ClickToPayWrapper` consumes the context values from the `ClickToPayProvider` through the `useClickToPayContext` hook . 

`ClickToPayProvider` uses the context do decide which payment method should be primary/secondary ones (Card or CTP)

Reply:
> The whole idea being that the CtP stuff is as self-contained & separated as possible, with the minimum disruption on the Card.tsx markup

I totally agree. In the current situation here, the 'ClickToPayWrapper' is acting mainly as a 'Card UI manager' in a sense that it is used to decide how to render the UI: only the 'card', only the 'loader', or the 'ctp + card' - it all depends of the 'state' of the Click to Pay (if it is available, initializing, etc)

I tried to encapsulate the logic in the Wrapper, so in case we provide a standalone ClickToPay component in the future, we won't have to mix the logic about Card component. We could simply do:

```jsx
<ClickToPayProvider clickToPayService={this.props.clickToPayService}>
    <ClickToPayComponent />
</ClickToPayProvider>
```

But I want to revisit this again and see if I can improve that more. I will do it after wrapping up the checkout flow.